### DISCUSSION TITLE
RaspberryMatic 3.73.9.20231130

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2539] [Announcements]
This is release 3.73.9.20231130 of RaspberryMatic which is a **major** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.73.9.20231130/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.73.9.20231130) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=80821)</sub>

## :construction: Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.71.12.20231020...3.73.9.20231130).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.73.9-1](https://github.com/jens-maus/occu/tree/b_3_73) integrating full compatibility to the [CCU3 3.73.9 firmware](https://homematic-ip.com/sites/default/files/other_media/ccu3-changelog-3-73-9.pdf) which comes with the following upstream (eQ3) performed changes:
  - `HmIP-BBL` - After a firmware update, the configuration page of the device could no longer be could no longer be called up.
  - `HmIP-FBL` - In the "Roller shutter" operating mode, the slat adjustment was activated with a long press of the button.
  - `Hm-Sen-MDIR-X` - Problems with the display of channel parameters fixed.
  - `HmIP-STE2-PCB (ELV kit)` - Incorrect designation of the configuration parameter "Minimum transmission distance" corrected, thus "Random component" was incorrectly displayed.
  - Java Runtime Environment updated to version 11.
- updated tailscale VPN to latest 1.54.0 version.
- Add port 2110 for homeputer Exec-Engine to HA-addon config (#2515, @tomquist)
- integrated fix in `eq3_char_loop` kernel module sources for compiler warning/error due to wrong printk format spec.

#### WebUI changes:
- modified 0170-WebUI-RegexSearchInDeviceLists WebUI patch to fallback to plain text search when regex is invalid (#2530, #2521, @IT-VBFK).
- added new 0182-WebUI-Fix-DimmerControl-InitButtonsWithChannelState WebUI patch which fixes unwanted delay in displaying the correct dimmer status (#1513, #2513, @jp112sdl).
- fixed issue in 0056-WebUI-Sticky-Filters which prevented that some sticky WebUI filter operations worked as expected (cf. #953).
- added usb device filesystem space stats to `help.cgi` and also query board vendor/name from dmi info.
- implemented DevConfig changes to fix device deletion and also present additional options to reset/force deletion as well as present device list and channel list slightly different (#2504).
- modified 0166-WebUI-Fix-PowermeterControl WebUI patch to show m3/h unit for the gas power unit which better fits the expected unit.
- added new 0181-WebUI-Fix-HTTPSfirmwareURLsOnly WebUI patch which makes sure that we only use https URLs for the device firmware update queries for improved security (#2491).

#### Operating system changes:
- bump rpi kernel to latest `stable_20231123` (6.1.63).
- updated upstream kernel to 6.1.63
- updated Azul-Java version to Java 11 (11.68.17-ca-jre11.0.21) as the HMIPServer of OCCU 3.73.x is requiring Java 11 itself.
- reworked whole kernel config management by prefering to use kconfig fragment files and kernel supplied defconfigs which are then overrided by the fragment files. This should make sure that future kernel updates should pickup new options/changes automatically. Furthermore, this allows to better group common kernel options in the "kernel" directory. Overall, this make the whole kernel config option management a bit more easier to maintain in future, especially if new supported platforms will be introduced.
- updated buildroot to latest 2023.08.3
- added `BR2_DOWNLOAD_FORCE_CHECK_HASHES` to make sure all package downloads are protected by proper hashes.
- explicitly disable the bcm283x serial driver support in the rpi versions of u-boot so that they won't interfere with our rpi-rf-mode dtbo. This should allow to remove the stdout-path workaround to prevent u-boot from crashing in case the rpi-rf-mod.dtbo is enabled in config.txt. (cf. https://github.com/home-assistant/operating-system/issues/2919)
- bump rpi-firmware to latest stable version.
- reworked the buildroot defconfig management to use fragment files for the boards buildroot config definitions rather than absolute defconfig files. This should make future adaptions to our buildroot environment easier.
- readded lost OK echo call on end of start() of eq3configd init script.
- moved seedrng init script to S05 and added /usr/local/etc dir check to make sure that upon a fresh factory reset seedrng is called correctly.
- modified logrotate config to allow users to put own logrotate config files in /usr/local/etc/logrotate.d so that the nightly logrotate call will also work over these user defined logrotate jobs. (cf. https://homematic-forum.de/forum/viewtopic.php?f=46&t=80457&p=783921#p783921).
- reworked `S01InitHost` to use `HWMODEL` rather than cpuinfo output to identify the RaspberryPi model. This should make ident of future models more easy.
- replaced all uboot defconfig files with kconfig fragment files which should make future U-Boot updates a bit easier and more secure because we reference the base defconfigs now that are directly shipped with U-Boot.
- modified `S01InitHost` to output the hardware model during bootup.
- updated u-boot Raspberry Pi patches to latest patchset.
- modified `82-hss_led.rules` udev rule to only match rpi_rf_mod led nodes and make sure to not modify permissions for all LED nodes.
- updated `qemu-guest-agent` to latest 8.1.3 and adapted build environment accordingly.

#### Misc changes:
- Update README of HA Addon with warning regarding the disabled UART (#2528, @majuss)

## :family: Contributors (alphabetically):
- @IT-VBFK
- @jens-maus
- @jp112sdl
- @majuss
- @tomquist

## :memo: Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## :package:  Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-rpi4.zip)<br/>SHA256: edaa2cefb037b57056816c8adadf9c3480a159db221fb18ebd0a73fd59a9d13b</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-rpi3.zip)<br/>SHA256: 768e86bcaa3e2ef717ac7ba2603d2a7a5f2791a85d1ba563eca26412865f1a74</sub>
<sub>:package: [RaspberryMatic-3.73.9.20231130-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: 448d7156863ef48a59c0543d5f079b6052080a5ec1a5ef97335587757135fe5f</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-rpi2.zip)<br/>SHA256: c1b14354d8e2313a4b48671cb95ce1f9f28046460308367aa3ed9e29ca807925</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-rpi0.zip)<br/>SHA256: 874f8f3c9667b0b8762f5e5c412db8bf0f3be08ba5d684bb4080624634c97fa2</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-tinkerboard.zip)<br/>SHA256: 119fdcd6cfc5b5490669462f2c74137cee864fbfd0156d2a926450bbac52923d</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-odroid-n2.zip)<br/>SHA256: 4cd1d182d5b8cf333634881336fbe2a2d8532599839aac3cb5da8ee1b4e08cd9</sub>
<sub>:package: [RaspberryMatic-3.73.9.20231130-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-odroid-c4.zip)<br/>SHA256: d74069ab693d6080f901759d7b0b3a3a7d777cbee584c1d5c3092bdcb6dd9410</sub>
<sub>:package: [RaspberryMatic-3.73.9.20231130-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-odroid-c2.zip)<br/>SHA256: c373b6ab02667999419d60148a1ce2e4168a0c9e9873080e0b2ca5f74904439e</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-intelnuc.zip)<br/>SHA256: 6e31df3b3eb07fc71f616e228016e49a176f4265eb54f0fb785d5b5606f83ee4</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-ova.zip)<br/>SHA256: 56a484b7aa890dca28899961ca85c3269b37d6da6bc87ae8596b28bffc743337</sub>
<sub>:package: [RaspberryMatic-3.73.9.20231130.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130.ova) (only for initial OVA installation)<br/>SHA256: 508951cf6a1c16defda75a9e70f96aad6647ad71fd8a5cc2cbadf9ebc1128528</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>:package: [RaspberryMatic-3.73.9.20231130-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: 56f674480919be100a363126602b17ec51883319e25f22b0f910d7887467d1fc</sub>
<sub>:package: [RaspberryMatic-3.73.9.20231130-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: 8ce2d82f27bbf8ab1ea172318849f9d8ee167a001b139f80709bb4be3ee3412c</sub>
<sub>:package: [RaspberryMatic-3.73.9.20231130-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20231130/RaspberryMatic-3.73.9.20231130-oci_arm.tgz) (arm/armhf)<br/>SHA256: 4fae528e79280b4e850513f5d3e6cd41eb0b687b549d205847163f7d44e1d062</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.73.9.20231130'>RaspberryMatic 3.73.9.20231130</a>.</em>

**Comment by sh0815907** [comment]
On Raspberry Zero 2 W the WiFi module does not work, so I have no network. On the previuos  version 3.71.12.20231020 it works.

**Reply by jens-maus** [reply]
If you have clear evidence for that (`/var/log/boot.log` and `dmesg` output not showing any `wlan0` interface) please open a dedicated issue ticket.

**Reply by sh0815907** [reply]
Logs from 3.73.9 on RPi Zero 2 W w/o working WLAN:
[boot-3.73.9.log](https://github.com/jens-maus/RaspberryMatic/files/13650930/boot-3.73.9.log)
[dmesg-3.73.9.txt](https://github.com/jens-maus/RaspberryMatic/files/13650932/dmesg-3.73.9.txt)

Logs from a 3.71.12 installation on a RPi Zero 2 W with working WLAN:
[boot-3.71.12.log](https://github.com/jens-maus/RaspberryMatic/files/13650935/boot-3.71.12.log)
[dmesg-3.71.12.txt](https://github.com/jens-maus/RaspberryMatic/files/13650936/dmesg-3.71.12.txt)

Thanks and hope this helps!
